File: 2492. Minimum Score of a Path Between Two Cities.java
class Solution {

    public int minScore(
            int numberOfCities,
            int[][] roads
    ) {

        // Build adjacency list
        List<List<int[]>> graph =
                new ArrayList<>();

        for (int i = 0;
             i <= numberOfCities;
             i++) {

            graph.add(
                    new ArrayList<>()
            );
        }

        for (int[] road : roads) {

            int cityA = road[0];
            int cityB = road[1];
            int roadDistance = road[2];

            graph.get(cityA).add(
                    new int[]{
                            cityB,
                            roadDistance
                    }
            );

            graph.get(cityB).add(
                    new int[]{
                            cityA,
                            roadDistance
                    }
            );
        }

        boolean[] visited =
                new boolean[numberOfCities + 1];

        Queue<Integer> bfsQueue =
                new LinkedList<>();

        bfsQueue.offer(1);
        visited[1] = true;

        int minimumScore =
                Integer.MAX_VALUE;

        // BFS traversal
        while (!bfsQueue.isEmpty()) {

            int currentCity =
                    bfsQueue.poll();

            for (int[] edge :
                    graph.get(currentCity)) {

                int nextCity = edge[0];
                int roadDistance = edge[1];

                // Update minimum edge weight
                minimumScore = Math.min(
                        minimumScore,
                        roadDistance
                );

                if (!visited[nextCity]) {

                    visited[nextCity] = true;

                    bfsQueue.offer(nextCity);
                }
            }
        }

        return minimumScore;
    }
}

File: 2492. Minimum Score of a Path Between Two Cities.py
from collections import deque
from typing import List

class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:

        # Build adjacency list
        graph = [[] for _ in range(n + 1)]

        for a, b, distance in roads:
            graph[a].append((b, distance))
            graph[b].append((a, distance))

        visited = [False] * (n + 1)

        q = deque()
        q.append(1)
        visited[1] = True

        minimum_score = float('inf')

        # BFS traversal
        while q:

            city = q.popleft()

            for next_city, distance in graph[city]:

                # update minimum edge in connected component
                minimum_score = min(minimum_score, distance)

                if not visited[next_city]:
                    visited[next_city] = True
                    q.append(next_city)

        return minimum_score
